Src/EngineIoClientDotNet.mono/Client/Socket.cs

```diff
@@ -445,7 +445,7 @@ private bool Flush()
             var log = LogManager.GetLogger(Global.CallerName());
 
             log.Info(string.Format("ReadyState={0} Transport.Writeable={1} Upgrading={2} WriteBuffer.Count={3}",ReadyState,Transport.Writable,Upgrading, WriteBuffer.Count));
-            if (ReadyState != ReadyStateEnum.CLOSED && this.Transport.Writable && !Upgrading && WriteBuffer.Count != 0)
+            if (ReadyState != ReadyStateEnum.CLOSED && ReadyState == ReadyStateEnum.OPEN && this.Transport.Writable && !Upgrading && WriteBuffer.Count != 0)
```

Comment:
this can be shortened:
if `ReadyState == ReadyStateEnum.OPEN` has to be true `ReadyState != ReadyStateEnum.CLOSED` is always true, the first condition is not needed